src/utils/MainApi.js

```diff
@@ -0,0 +1,82 @@
+class MainApi {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse = async (res) => {
+    return res.ok ? res.json() : Promise.reject(await res.json());
+  };
+
+  saveArticle(data) {
+    return fetch(`${this._baseUrl}/articles/`, {
+      method: "POST",
+      headers: this._headers,
+      mode: "cors",
+      body: JSON.stringify(data),
+    }).then(this._checkResponse);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.